lighthouse.go

```diff
@@ -630,13 +634,18 @@ func (lh *LightHouse) addCalculatedRemotes(vpnAddr netip.Addr) bool {
 // unlockedGetRemoteList
 // assumes you have the lh lock
 func (lh *LightHouse) unlockedGetRemoteList(allAddrs []netip.Addr) *RemoteList {
-	am, ok := lh.addrMap[allAddrs[0]]
-	if !ok {
-		am = NewRemoteList(allAddrs, func(a netip.Addr) bool { return lh.shouldAdd(allAddrs[0], a) })
-		for _, addr := range allAddrs {
-			lh.addrMap[addr] = am
+	// before we go and make a new remotelist, we need to make sure we don't have one for any of this set of vpnaddrs yet
+	for _, addr := range allAddrs {
+		am, ok := lh.addrMap[addr]
```

Comment:
in the case where there is a remotelist for `allAddrs[1]` and not for `allAddrs[0]`, this will return the remotelist for `allAddrs[1]` and _not_ create an index in `addrMap` for `allAddrs[0]`. As I'm poking around, I can't tell where we'd ever fill it in if it didn't exist yet (which is a condition `static_host_map` could cause).